### PR TITLE
bug fix-Matter GUI

### PR DESCRIPTION
**Pull request by paulr34**
Matter GUI should allow client clusters but not show attributes or commands if a client cluster

### CODE REVIEW

File: src/components/ZclAttributeManager.vue [hunk 1]
```diff
@@ -39,7 +39,11 @@ limitations under the License.
         <q-tr
           :props="props"
           class="table_body"
-          v-if="!globalLists.includes(props.row.label)"
+          v-show="
+            (!globalLists.includes(props.row.label) &&
+              props.row.side === 'server') ||
```
Comment (brdandu) [inline src/components/ZclAttributeManager.vue:44]: This should be matter only change. Other protocols such as Zigbee should not be affected by this. Is that the case here?

File: cypress/e2e/attributes/check-required-attributes.cy.js [hunk 2]
```diff
@@ -16,22 +16,4 @@ describe('Testing disabling enabled attributes', () => {
       cy.gotoAttributePage(data.endpoint1, data.cluster1)
     })
   })
-  it(
```
Comment (brdandu) [inline cypress/e2e/attributes/check-required-attributes.cy.js:19]: Why is this UI test removed?

File: src/components/ZclAttributeReportingManager.vue [hunk 3]
```diff
@@ -29,7 +29,10 @@ limitations under the License.
       data-cy="Attributes Reporting"
     >
       <template v-slot:body="props">
-        <q-tr :props="props">
+        <q-tr
+          :props="props"
+          v-show="props.row.side === 'server' || !enableServerOnly"
+        >
```
Comment (brdandu) [inline src/components/ZclAttributeReportingManager.vue:35]: Does matter have attribute reporting? Should this whole vue component not show up for matter?

File: src/components/ZclCommandManager.vue [hunk 4]
```diff
@@ -35,7 +35,10 @@ limitations under the License.
           </q-tr>
         </template>
         <template v-slot:body="props">
-          <q-tr :props="props">
+          <q-tr
+            :props="props"
+            v-show="props.row.source === 'server' || !enableServerOnly"
```
Comment (brdandu) [inline src/components/ZclCommandManager.vue:40]: Should we just be showing commands which have source as server?
Are there no response commands which should be shown when command source is client?

**Review by brdandu — CHANGES_REQUESTED** [review]

File: src/components/ZclAttributeManager.vue (hunk 1 above)
Comment (paulr34) [reply src/components/ZclAttributeManager.vue:44]: it does do that `            (!globalLists.includes(props.row.label) &&
              props.row.side === 'server') ||
            !enableServerOnly`

File: cypress/e2e/attributes/check-required-attributes.cy.js (hunk 2 above)
Comment (paulr34) [reply cypress/e2e/attributes/check-required-attributes.cy.js:19]: its checking for a client side attribute which means it is an incorrectly written test and should be remade in a separate PR

File: src/components/ZclAttributeReportingManager.vue (hunk 3 above)
Comment (paulr34) [reply src/components/ZclAttributeReportingManager.vue:35]: good point, ill add that

File: src/components/ZclCommandManager.vue (hunk 4 above)
Comment (paulr34) [reply src/components/ZclCommandManager.vue:40]: I thought that it was discussed we don't need to show Client commands but not 100 percent sure

File: cypress/e2e/attributes/check-required-attributes.cy.js (hunk 2 above)
Comment (brdandu) [reply cypress/e2e/attributes/check-required-attributes.cy.js:19]: Shouldn't it still be tested for protocols other than matter?

**Comment by paulr34** [comment]
going with a different design

**Pull request closed without merging**